Statistics/Distributions/Deterministic.cs

```diff
@@ -1,56 +1,47 @@
-﻿using System;
+﻿using Base.Implementations;
+using Base.Enumerations;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using System.Xml.Linq;
 using Utilities;
-using Utilities.Serialization;
 
 namespace Statistics.Distributions
 {
-    public class Deterministic : IDistribution
+    public class Deterministic : ContinuousDistribution
     {
         #region IDistribution Properties
-        public IDistributionEnum Type => IDistributionEnum.Deterministic;
-
-        public double Mean { get; set;  }
-
-        public double Median { get; set; }
-
-        public double Mode { get; set; }
-
-        public double Min { get; set; }
-        public double Max { get; set; }
-        public double Variance { get; set; }
-
-        public double StandardDeviation { get; set; }
-
-        public double Skewness { get; set; }
-
+        public override IDistributionEnum Type => IDistributionEnum.Deterministic;
         public IRange<double> Range { get; set; }
 
-        public int SampleSize { get; set; }
-        public bool Truncated { get; set; }
-        public IMessageLevels State => throw new NotImplementedException();
+        public override int SampleSize { get; protected set; }
```

Comment:
can you please tell me about why we need override here?

Reply:
because i marked it as abstract in the parent class ContinuousDistribution. An alternative would be to implement it in ContinuousDistribution - which I might do before we merge this branch. That way all concrete implementations dont need this boiler plate.